api/src/main/java/org/apache/iceberg/actions/GenerateSymlinkManifest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+/**
+ * An action that generates a symlink manifest table for an Iceberg table.
+ *
+ * <p>This action will generate symlink manifests for a table. A symlink manifest is a file with
+ * links to the data files in the table. Systems which do not integrate with Iceberg can read the
```

Comment:
Nit: This paragraph is correct in explaining what a symlink manifest is, but I do encounter a number of people on Slack / in the community who aren’t aware that Icebergs metadata tracks the files already. I worry this will make that _less_ clear.

Is it possible to document the symlink manifest without making it seem like Iceberg doesn’t already track the files? Maybe something like the following would help.

`A symlink manifest is a standardized table format (specification file?) that many other systems can read. It consists of links to the data files in the table, like Iceberg, but without the additional metadata that Iceberg includes. Systems which do not integrate with Iceberg can read the …`

Not necessarily in love with that phrasing, but given how often we wind up fielding questions from users about performance due to table layout etc because newer users aren’t always aware that Iceberg tracks the files in its manifests, it would be nice to make that a little more explicit somehow so as not to potentially introduce more confusion. Again, not blocking as this is a good description but I know we both answered a question about table layout on Slack just this week for a user who was assuming that Iceberg was doing listing to get partition info. It would be good to reduce the chances of getting more of these questions if possible. 🙂